lib/tasks/TaskScheduler.js

```diff
@@ -0,0 +1,91 @@
+const async = require('async');
+
+/**
+ * Schedule replication tasks according to:
+ *
+ * - An optional queue key that guarantees all tasks sharing the same
+ *   queue key will be processed in order and serially
+ *
+ * - An optional dedupe key that will skip new tasks if they share the
+ *   same dedupe key than another task either in progress or already
+ *   queued
+ */
+class ReplicationTaskScheduler {
```

Comment:
~ReplicationTaskScheduler~ TaskScheduler

Looks like we still need to update the name and path across files.

Reply:
Yes, I just forgot to commit that bit originally :rofl: